### DISCUSSION TITLE
When function "Simplifying Polygonal Coverages" from JTS / GEOS available?

### DISCUSSION THREAD

**Discussion by jome1** [shapely/shapely#2080] [Q&A] [answered]
Dear community,

last year the JTS Topology Suite got a new function, which allows to simplify adjacent polygons. See [this blog post](https://lin-ear-th-inking.blogspot.com/2023/03/simplifying-polygonal-coverages-with-jts.html) for a really nice explanation. `Shapely` is used in one of the [tools](https://github.com/pypsa-meets-earth/pypsa-earth) I want to use and this tool would tremondously benefit from this new function.
In the blogpost it is said that: _"As usual, this algorithm will be ported to [GEOS](http://libgeos.org/), from where it will be available to downstream projects such as [PostGIS](http://www.postgis.org/) and [Shapely](https://github.com/shapely/shapely)."_

So is it somehow possible to tell a timeline when this function will be available?

Thx!

**Comment by martinfleis** [comment, marked as answer]
There is a PR adding this function in #1969 so I suppose it will be part of 2.1 release, which does not yet have a release date. In the meantime, you should be able to build shapely from a branch linked to that PR if you need to use it.

**Reply by jome1** [reply]
Oh that's awesome! Thank you!

**Comment by theroggy** [comment]
If your data/coverage you want to simplify is not a valid coverage (topologically sound) yet, you could also consider looking at https://github.com/mattijn/topojson.

You can use that library to do a topological/coverage simplify as well ([toposimplify](https://mattijn.github.io/topojson/example/settings-tuning.html#toposimplify)), and it will do some basic things to try to fix some obvious topological issues before doing so. There are plans to add functionalities to fix topological issues in GEOS as well, but they are not developed yet...
